redoing bedrock docs + codeowner change (#14008)
cloudflare/cloudflare-docs: content/ai-gateway/_partials/_bedrock.md

`https://gateway.ai.cloudflare.com/v1/ACCOUNT_TAG/GATEWAY/aws-bedrock`

When making requests to Amazon Bedrock, replace `https://bedrock-runtime.us-east-1.amazonaws.com/` in the URL you’re currently using with `https://gateway.ai.cloudflare.com/v1/ACCOUNT_TAG/GATEWAY/aws-bedrock/bedrock-runtime/us-east-1/`, then add the model you want to run at the end of the URL.

With Bedrock, you'll need to sign the URL before you make requests to AI Gateway. You can try using the [aws4fetch](https://github.com/mhart/aws4fetch) sdk like below.

> ```javascript

Reviewer: ```suggestion
```typescript
```

import { AwsClient } from 'aws4fetch'

export default {

>     async fetch(request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {

Reviewer: `Env` tyoe isn't defined anywhere here. Is this meant to come from somewhere?

        // replace with your configuration

>         const endpoint = 'https://gateway.ai.cloudflare.com/v1/ACCOUNT_ID/GATEWAY/aws-bedrock/bedrock-runtime/REGION/model/amazon.titan-embed-text-v1/invoke';

Reviewer: This variable isn't used?

        const accessKey = 'removed';

>         const secretKey = 'removed';

Reviewer: Should we encourage users store keys like this in plaintext within the worker? [Secrets](https://developers.cloudflare.com/workers/configuration/secrets/) would be better I'd think?

        const region = 'us-east-1';

        const requestData = {

>                 inputText: "What does ethereal mean?"

Reviewer: This and other objects are indented with 8 spaces, whereas most other indentation is 4 spaces. Some consistency would probably be ideal.

        };

        const headers = {
                'Content-Type': 'application/json'
        };

        // sign the original request

>         const stock_url = new URL("https://bedrock-runtime.us-east-1.amazonaws.com/model/amazon.titan-embed-text-v1/invoke")

Reviewer: nitipick: most other variables here are camelCase, whereas this one is snake_case. Some consistency would probably be ideal.

        const awsClient = new AwsClient({
                accessKeyId: accessKey,
                secretAccessKey: secretKey,
                region: region,
                service: "bedrock"
        })

        const presigned_request = await awsClient.sign(stock_url.toString(), {
                method: "POST",
                headers: headers
        });

        // change the signed request's host to AI Gateway 
        const stock_url_signed = new URL(presigned_request.url);
        stock_url_signed.host = "gateway.ai.cloudflare.com"

>         stock_url_signed.pathname = "/v1/ACCOUNT_ID/GATEWAY/aws-bedrock/bedrock-runtime/REGION/model/amazon.titan-embed-text-v1/invoke"

Reviewer: is this meant to use the `endpoint` variable in some way so users don't have to specify their `ACCOUNT_ID`, `GATEWAY`, and `REGION` twice?

        // make request
        const response = await fetch(stock_url_signed, {
            method: 'POST',
            headers: presigned_request.headers,
            body: JSON.stringify(requestData)
        })

        const data = await response.json();

>         return new Response(data);

Reviewer: This won't work. `return new Response({foo:'bar'})` will simply return `[object Object]`.

You probably want to ``JSON.stringify` or `Response.json`. But that assumes this is always JSON, and won't respond with anything else. Checking the status code and content type is probably a good idea before doing this.

`https://gateway.ai.cloudflare.com/v1/ACCOUNT_TAG/GATEWAY/aws-bedrock`

When making requests to Amazon Bedrock, replace `https://bedrock-runtime.us-east-1.amazonaws.com/` in the URL you’re currently using with `https://gateway.ai.cloudflare.com/v1/ACCOUNT_TAG/GATEWAY/aws-bedrock/bedrock-runtime/us-east-1/`, then add the model you want to run at the end of the URL.

> With Bedrock, you'll need to sign the URL before you make requests to AI Gateway. You can try using the [aws4fetch](https://github.com/mhart/aws4fetch) sdk like below.

Reviewer 2: ```suggestion
With Bedrock, you will need to sign the URL before you make requests to AI Gateway. You can try using the [`aws4fetch`](https://github.com/mhart/aws4fetch) SDK. For example:
```

```typescript
import { AwsClient } from 'aws4fetch'

> inferface Env {
>   const accessKey: string;
>   const secretAccessKey: string;
> }

Reviewer: ```suggestion
inferface Env {
  accessKey: string;
  secretAccessKey: string;
}
```

Reviewer 3: D'oh! Thanks for catching!

    const accessKey = env.accessKey;
    const secretKey = env.secretAccessKey;

    const requestData = {
      inputText: "What does ethereal mean?"
    };

    const headers = {
      'Content-Type': 'application/json'
    };

    // sign the original request
    const stockUrl = new URL("https://bedrock-runtime.us-east-1.amazonaws.com/model/amazon.titan-embed-text-v1/invoke")

    const awsClient = new AwsClient({
      accessKeyId: accessKey,
      secretAccessKey: secretKey,
      region: region,
      service: "bedrock"
    });

    const presignedRequest = await awsClient.sign(stockUrl.toString(), {
      method: "POST",
      headers: headers
    });

    // change the signed request's host to AI Gateway
    const stockUrlSigned = new URL(presignedRequest.url);
    stockUrlSigned.host = "gateway.ai.cloudflare.com"
    stockUrlSigned.pathname = `/v1/${cfAccountId}/${gatewayName}/aws-bedrock/bedrock-runtime/${region}/model/amazon.titan-embed-text-v1/invoke`

    // make request
    const response = await fetch(stockUrlSigned, {
      method: 'POST',
      headers: presignedRequest.headers,
      body: JSON.stringify(requestData)
    })

>     if (response.ok && response.headers.get('content-type').includes('application/json')) {

Reviewer 3: Is this too naive / clunky?

Reviewer: This could fail if `content-type` isn't set on the response, and therefore returns `null`, so you'll get:
```
Uncaught TypeError: Cannot read properties of null (reading 'includes')
```

An easy fix would be this (optional chaining after the header get):

```suggestion
    if (response.ok && response.headers.get('content-type')?.includes('application/json')) {
```
